tests/foreman/api/test_permission.py

```diff
@@ -200,11 +200,7 @@ def give_user_permission(self, perm_name):
 
         """
         role = entities.Role().create()
-        permissions = [
-            entities.Permission(id=permission['id'])
-            for permission
-            in entities.Permission(name=perm_name).search()
-        ]
+        permissions = entities.Permission(name=perm_name).search()
```

Comment:
i believe you removed the entity's old search capabilities in the new mixin PR so no worries here!